Lib9c/Module/AvatarModule.cs

```diff
@@ -119,160 +127,44 @@ public static AvatarState GetEnemyAvatarState(IWorldState worldState, Address av
-            IWorldState worldState,
-            Address agentAddress,
-            Address avatarAddress,
-            out AvatarState avatarState,
-            out bool migrationRequired
-        )
-        {
-            avatarState = null;
-            migrationRequired = false;
-            if (AccountHelper.Resolve(worldState, avatarAddress, Addresses.Avatar) is Dictionary
-                serializedAvatar)
-            {
-                try
-                {
-                    if (serializedAvatar[AgentAddressKey].ToAddress() != agentAddress)
-                    {
-                        return false;
-                    }
-
-                    avatarState = GetAvatarStateV2(worldState, avatarAddress);
-                    return true;
-                }
-                catch (Exception e)
-                {
-                    // BackWardCompatible.
-                    if (e is KeyNotFoundException || e is FailedLoadStateException)
-                    {
-                        migrationRequired = true;
-                        return TryGetAvatarState(
-                            worldState,
-                            agentAddress,
-                            avatarAddress,
-                            out avatarState);
-                    }
-
-                    return false;
-                }
-            }
-
-            return false;
-        }
-
-        // FIXME: Should not use this unified method.
-        public static bool TryGetAgentAvatarStates(
-            IWorldState worldState,
-            Address agentAddress,
-            Address avatarAddress,
-            out AgentState agentState,
-            out AvatarState avatarState
-        )
-        {
-            avatarState = null;
-            agentState = AgentModule.GetAgentState(worldState, agentAddress);
-            if (agentState is null)
+            catch (Exception)
             {
                 return false;
             }
-
-            if (!agentState.avatarAddresses.ContainsValue(avatarAddress))
-            {
-                throw new AgentStateNotContainsAvatarAddressException(
-                    $"The avatar {avatarAddress.ToHex()} does not belong to the agent {agentAddress.ToHex()}.");
-            }
-
-            avatarState = GetAvatarState(worldState, avatarAddress);
-            return !(avatarState is null);
         }
 
-        // FIXME: Should not use this unified method.
-        public static bool TryGetAgentAvatarStatesV2(
-            IWorldState worldState,
-            Address agentAddress,
+        public static IWorld SetAvatarState(
+            IWorld world,
             Address avatarAddress,
-            out AgentState agentState,
-            out AvatarState avatarState,
-            out bool avatarMigrationRequired
-        )
+            AvatarState state,
+            bool setAvatar,
+            bool setInventory,
+            bool setWorldInformation,
+            bool setQuestList)
```

Comment:
Actually, I think these boolean options are just used for tests. May be it's better to update tests to use the most recent version, and remove theses options, if we have enough time.

Reply:
Oops, my mistake. these have to be remained, to prevent values not updated to be set.

Reply:
And these seems to be removed, since we're splitting those to other accounts.